mypy/docstrings.py

```diff
@@ -0,0 +1,27 @@
+from typing import List, Tuple, Dict
+from mypy.types import Type, CallableType, AnyType
+from mypy.nodes import Argument
+
+
+def parse_docstring(docstring: str, line: int) -> Tuple[Dict[str, Type], Type]:
+    """
+    Parse a docstring and return type representations.  This function can
+    be overridden by third-party tools which aim to add typing via docstrings.
+
+    Returns a 2-tuple: dictionary of arg name to Type, and return Type.
```

Comment:
> I wonder if Type is the right thing to return here. Turning a string representing a type as found in the source code into a Type object is a fairly complicated process, and we already have code for that (the code that parses type comments). 

I'm using `mypy.parsetype.parse_str_as_type` to handle the conversion within my little experimental project, so it's pretty straight-forward.  Returning strings means that mypy will have to have more of an opinion about how to deal with error handling and such, but it would avoid the need for the `parse_docstring` function to take a line number (which is there explicitly for passing to `parse_str_as_type`).  I'll look into whether I can return `Dict[str, str]` while still maintaining the current functionality in my project.

> As a nit, I'd use the same convention for the return type as used in __annotations__ and a few of inspect's APIs -- just use a key "return" since that can't be an argument name anyway.

Yeah, I like that.  It's a simpler structure to document.